components/schema-migrator/migrations/director/20211111122118_bundles_correlation_id.up.sql

```diff
@@ -0,0 +1,81 @@
+BEGIN;
+
+ALTER TABLE bundles ADD COLUMN correlation_ids JSONB;
+
+DROP VIEW IF EXISTS tenants_bundles;
+DROP VIEW IF EXISTS correlation_ids;
+
+CREATE OR REPLACE VIEW tenants_bundles
+            (tenant_id, provider_tenant_id, id, app_id, name, description, instance_auth_request_json_schema,
+             default_instance_auth, ord_id,
+             short_description, links, labels, credential_exchange_strategies, ready, created_at, updated_at,
+             deleted_at, error, correlation_ids)
+AS
+SELECT DISTINCT t_apps.tenant_id,
+                t_apps.provider_tenant_id,
+                b.id,
+                b.app_id,
+                b.name,
+                b.description,
+                b.instance_auth_request_json_schema,
+                b.default_instance_auth,
+                b.ord_id,
+                b.short_description,
+                b.links,
+                b.labels,
+                b.credential_exchange_strategies,
+                b.ready,
+                b.created_at,
+                b.updated_at,
+                b.deleted_at,
+                b.error,
+                b.correlation_ids
+FROM bundles b
+         JOIN (SELECT a1.id,
+                      a1.tenant_id::text,
+                      a1.tenant_id::text AS provider_tenant_id
+               FROM applications a1
+               UNION ALL
+               SELECT a.id,
+                      t.parent::text AS tenant_id,
+                      t.parent::text AS provider_tenant_id
+               FROM applications a
+                        JOIN business_tenant_mappings t ON t.id = a.tenant_id
+               WHERE t.parent IS NOT NULL
+               UNION ALL
+               SELECT *
+               FROM apps_subaccounts_func()
+               UNION ALL
+               SELECT a_s.id, (SELECT id::text FROM business_tenant_mappings WHERE external_tenant = a_s.tenant_id::text), cpr.provider_tenant::text AS provider_tenant_id
+               FROM apps_subaccounts_func() a_s
+                        JOIN consumers_provider_for_runtimes_func() cpr
+                             ON cpr.consumer_tenants ? a_s.tenant_id::text) t_apps ON b.app_id = t_apps.id;
+
+
+CREATE OR REPLACE VIEW correlation_ids(application_id, product_id, bundle_id, value) as
```

Comment:
I think the original `correlation_ids` (latest version can be seen [here](https://github.com/kyma-incubator/compass/blob/main/components/schema-migrator/migrations/director/20210622095252_vendor_product_internal_ids.up.sql#L148)) should be extended with the new bundle logic.

My main concern with the current approach is that the newly introduced `LATERAL` functionality does not add much value because the view is not complex at all.

Moreover, I find the aliases at the beginning: `SELECT app_correlation_ids.product_id, app_correlation_ids.bundle_id` very confusing because product and bundle ids have nothing to do with the app_correlation_ids.

The down migration should use the original `correlation_ids` view as well.